build_docs.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+# Build the docs with docker!
+#
+# Step 1 is to build a docker image based on the asciidoctor image.
+# Step 2 is to translate the arguments that build_docs.pl supports into
+# a list of arguments to be passed to start the docker container and a
+# list of arguments to be passed to the build_docs.pl process that is
+# started in the docker container.
+# Step 3 is to start the docker container. We start it in such a way
+# that is *should* remove itself when it is done.
+
+set -e
+
+function desymlink() {
+  FILE="$1"
```

Comment:
Is this an implementation of `realpath(1)`? OSX doesn't have that, huh? Wowsers.

It does have Python with `os.path.realpath()`, though (see above). Python (or other modern scripting language) would eliminate a bunch of these disparities that work differently between GNU and BSD.

I assume you can't dereference symlinks like this on OSX either:

```
$ stat -c%N /tmp/link | cut -d "'" -f4
/tmp/i/am/a/symlinked/file.txt
```

I feel like Bash on BSD is really punishing you here if it's making you write `realpath(1)`! Philosophically, there's lot of system-dependant code to get us into Docker so we can stop thinking about the system.

Reply:
But you can, according to the manpage, do `stat -c%Y /the/link`, which is _really_ nice but not in GNU. Arrggggg!